### ISSUE TITLE
Default for AvailabilityGroupFileName

### ISSUE THREAD

**Issue by SlavaSQL** [olahallengren/sql-server-maintenance-solution#334]
In the current settings FileName and AvailabilityGroupFileName have to be set differently and they have different defaults. If you set @FileName, it does not affect @AvailabilityGroupFileName and it still go with the default value.
That means if you want your file names to look the same for all databases, no matter if they are in AG or not, you have to specify BOTH parameter.

It might be nice to have a rule for @AvailabilityGroupFileName like this:
1. If it is specified then it goes as specified.
2. If it is not specified and @FileName is not specified then it goes as default.
3. If it is not specified, but @FileName IS specified then default value for @AvailabilityGroupFileName will be set the same as @FileName.

The idea behind: 
1. Specify only one file name parameter if you need names be the same.
2. Both names will be always the same if @FileName is specified and @AvailabilityGroupFileName is not specified

Caveats:
Only possible problem I see is a situation when somebody wants following:
1. Want their own standard for @FileName
2. Want to have default value for @AvailabilityGroupFileName

**Comment by olahallengren** [comment]
Let's start with a background for the parameters for the directory structure.

If you have availability groups and you are backing up to an UNC path, then it makes restore easier, if you have the backups in the same directory, after a failover. That's why I use the cluster name and availability group name, by default in the directory path for databases in availability groups.

For file names I follow the same pattern. It is nice to be able to see that some backup files are from the same cluster and availability group.

**Comment by SlavaSQL** [comment]
Sorry for the delay. Missed the response.
I absolutely agree with you on the Directory structure, but filenames do not to be different, because you already know they are in the AG folder.

I know it is easy fix, but at first it brings a confusion to the users. They thought it is an error and only after I've researched and found the additional parameter, they could use unified file names.

I assume when you have files in a repository, you do not care if they belong to a database in AG or not. In case of a disaster it won't matter.

I'm not against that different file naming for AG backups, but just think it would be more logical to have a default names unified by default.

**Comment by olahallengren** [comment]
I think that for databases in availability groups, the cluster name and availability group name is the most important information. The server name and the instance name are less important, as it does not matter on which replica a backup has been performed.

In any case, the parameters supports all configurations. I am not planning to make these parameters dependent on each other.

**Comment by SlavaSQL** [comment]
I understand.
